src/Datadog.Trace/Span.cs

```diff
@@ -359,8 +359,14 @@ public void SetException(Exception exception)
         /// <returns> The value for the tag with the key specified, or null if the tag does not exist</returns>
         public string GetTag(string key)
         {
-            // no need to lock on single reads
-            return Tags != null && Tags.TryGetValue(key, out var value) ? value : null;
+            switch (key)
+            {
+                case Trace.Tags.SamplingPriority:
+                    return Context.TraceContext.SamplingPriority.ToString();
```

Comment:
This probably needs to copy the logic from [SpanContextPropagator](https://github.com/DataDog/dd-trace-dotnet/blob/7f468ad26fb0be0921326a2fd253a887ea7eb7b3/src/Datadog.Trace/SpanContextPropagator.cs#L40).

Reply:
- the sampling priority can be stored in two places (`context.TraceContext.SamplingPriority` or `context.SamplingPriority`)
- we should propagate this as the numeric value (`"1"`, `"2"`, etc) so it's compatible across language tracers